components/shared/embedder/viewport_description.rs

```diff
@@ -0,0 +1,144 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! This module contains helpers for Viewport
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use style_traits::PinchZoomFactor;
+
+/// A set of values for Length:
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum ViewportLength {
+    /// Length is auto
+    Auto,
+    /// Length is device width
+    DeviceWidth,
+    /// Length is device height
+    DeviceHeight,
+    /// Length is fixed
+    Fixed(f32),
+}
+
+/// A set of User Zoom values:
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum UserZoom {
+    /// Zoom is not allowed
+    Fixed = 0,
+    /// Zoom is allowed
+    Zoom = 1,
+}
+
+/// Default viewport constraints
+const MIN_ZOOM: f32 = 0.1;
+const MAX_ZOOM: f32 = 10.0;
+const DEFAULT_ZOOM: f32 = 1.0;
+
+/// A set of viewport descriptors:
+///
+/// <https://www.w3.org/TR/css-viewport-1/#viewport-meta>
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct ViewportDescription {
+    /// width
+    pub width: ViewportLength,
```

Comment:
Since width and height are not consumed, could you exclude them from this PR, and instead leave a TODO for them?